File: RB101_Programming_Foundations/small_problems/Easy5/Easy5_Python/letter_counter2.py
# Write a method that takes a string with one or more space separated words and returns a hash that shows the number of words of different sizes.

# Words consist only alphabet characters that do not include a space.

'''
Input: a string

Output: a dictinoary with keys the length of words, and values the occurances of that length

Rules


Data structure / Algorithms
    split the sentence into words by space
    create an empty hash
    iterate through the words
        remove the word of non-alphabet characters
        set the hash's key to the length of the current word, and value += 1, starting at 0
    return the hash

'''

def word_sizes(string):
    words = string.split()
    hash = {}

    for word in words:
        word_length = 0
        for char in word:
            if char.isalpha():
                word_length += 1

        hash[word_length] = hash.setdefault(word_length, 0) + 1

    return hash




print(word_sizes('Four score and seven.') == { 3 : 1, 4 : 1, 5 : 2 })
print(word_sizes('Hey diddle diddle, the cat and the fiddle!') == { 3 : 5, 6 : 3 })
print(word_sizes("What's up doc?") == { 5 : 1, 2 : 1, 3 : 1 })
print(word_sizes('') == {})

File: js210_fundamentals_of_JS_Programming/problem_sets/small_problems/Easy3/letter_counter2.js
function wordSizes(string) {
  if (!string) return {};

  return string.split(' ').reduce((hash, word) => {
    let length = word.replace(/[^a-z]/gi, '').length;
    hash[length] = (hash[length] || 0) + 1;
    return hash;
  }, {});
}

console.log(wordSizes('Four score and seven.')); // { "3": 1, "4": 1, "5": 2 }
console.log(wordSizes('Hey diddle diddle, the cat and the fiddle!')); // { "3": 5, "6": 3 }
console.log(wordSizes("What's up doc?")); // { "5": 1, "2": 1, "3": 1 }
console.log(wordSizes('')); // {}
